### PR TITLE
Features/7596

### PR DESCRIPTION
**Pull request by sslaws**
Initial work on deleting old records from file loads.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_175)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_175)

File: Apps/DBMaintainer/Apps/BCProvDrugDBApp.cs [hunk 1]
```diff
@@ -53,13 +53,16 @@ protected override string configurationName
...
+            downloadedFile.ProgramTypeCodeId = Database.Constant.ProgramType.Provincial;
             string[] files = Directory.GetFiles(sourceFolder, "pddf*.csv");
             if (files.Length > 1)
             {
                 throw new ApplicationException($"The zip file contained {files.Length} CSV files, very confused.");
             }
             this.logger.LogInformation("Parsing Provincial PharmaCare file");
             List<PharmaCareDrug> pharmaCareDrugs = this.parser.ParsePharmaCareDrugFile(files[0]);
+            // inject the FileDownload into the pharmacare drugs
+            pharmaCareDrugs.ForEach(c => c.FileDownload = downloadedFile);
```
Comment (FuriousLlama) [inline Apps/DBMaintainer/Apps/BCProvDrugDBApp.cs]: Instead of iterating over the new entries, how about passing the fileDownload to the parser?

Comment (sslaws) [reply Apps/DBMaintainer/Apps/BCProvDrugDBApp.cs]: I've changed this as discussed.

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_175)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_175)

**Comment by healthgatewaybc** [comment]
Database Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=databaselibrary_pr_175)

**Pull request merged**